### PR TITLE
e2e: shuffle tests and build with `-race`

### PR DESCRIPTION
**Pull request by Choraden**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,5 @@
-FROM gcr.io/distroless/static:nonroot
+ARG BASE_IMAGE=gcr.io/distroless/static:nonroot
```
Comment (mmatczuk) [inline Dockerfile:1]: Not used.

File: Makefile [hunk 2]
```diff
@@ -71,10 +71,14 @@ coverage:
 .PHONY: update-devel-image
 update-devel-image: TAG=devel
 update-devel-image: TMPDIR:=$(shell mktemp -d)
+update-devel-image: CI_OR_LINUX:=$(or $(CI),$(findstring Linux,$(shell uname)))
+update-devel-image: RACE_FLAG:=$(if $(CI_OR_LINUX),-race,)
+update-devel-image: CGO_ENABLED:=$(if $(CI_OR_LINUX),1,0)
+update-devel-image: IMAGE:=$(if $(CI_OR_LINUX),ubuntu:latest,gcr.io/distroless/static:nonroot)
```
Comment (mmatczuk) [inline Makefile:77]: I was hoping to see mk if branches not that. 

```
ifeq ($(shell uname),Linux)
```

**Review by mmatczuk — CHANGES_REQUESTED** [review]

File: Makefile [hunk 3]
```diff
@@ -72,9 +72,14 @@ coverage:
 update-devel-image: TAG=devel
 update-devel-image: TMPDIR:=$(shell mktemp -d)
 update-devel-image:
-	@CGO_ENABLED=0 GOOS=linux go build -o $(TMPDIR)/forwarder ./cmd/forwarder
 	@ln Dockerfile LICENSE LICENSE.3RD_PARTY $(TMPDIR)
-	@docker buildx build -t saucelabs/forwarder:$(TAG) $(TMPDIR)
+ifeq ($(shell uname),Linux)
+	@CGO_ENABLED=1 GOOS=linux go build -race -o $(TMPDIR)/forwarder ./cmd/forwarder
+	@docker buildx build --build-arg BASE_IMAGE=ubuntu:latest -t saucelabs/forwarder:$(TAG) $(TMPDIR)
```
Comment (mmatczuk) [inline Makefile:78]: Why base image is different?

Comment (Choraden) [reply Makefile:78]: distroless:static does not have libc

Comment (mmatczuk) [reply Makefile:78]: Gotcha how about `base`

https://github.com/GoogleContainerTools/distroless/blob/main/base/README.md

Comment (Choraden) [reply Makefile:78]: Base image also had some issues. 
You can check the pipeline with base image here: https://github.com/saucelabs/forwarder/actions/runs/6932307079/job/18855937399

**Pull request merged**